airflow-core/src/airflow/api_fastapi/execution_api/routes/dag_runs.py

```diff
@@ -150,3 +152,27 @@ def get_dagrun_state(
         )
 
     return DagRunStateResponse(state=dag_run.state)
+
+
+@router.get("/count", status_code=status.HTTP_200_OK)
+def get_dr_count(
+    dag_id: str,
+    session: SessionDep,
+    logical_dates: Annotated[list[UtcDateTime] | None, Query()] = None,
+    run_ids: Annotated[list[str] | None, Query()] = None,
+    states: Annotated[list[str] | None, Query()] = None,
+) -> int:
+    """Get the count of DAG runs matching the given criteria."""
+    query = select(func.count()).select_from(DagRun).where(DagRun.dag_id == dag_id)
+
+    if logical_dates:
+        query = query.where(DagRun.logical_date.in_(logical_dates))
+
+    if run_ids:
+        query = query.where(DagRun.run_id.in_(run_ids))
+
+    if states:
+        query = query.where(DagRun.state.in_(states))
+
+    count = session.scalar(query)
+    return count or 0
```

Comment:
We will need to add cadwyn migration for the new endpoints: https://docs.cadwyn.dev/concepts/version_changes/

Reply:
I don't think so. We only need to add a migration for breaking changes (or changes to existing endpoints) from what I understand.

Reply:
Just checked it here: https://docs.cadwyn.dev/concepts/endpoint_migrations/#defining-endpoints-that-didnt-exist-in-old-versions. Seems we will need it. Let me take it up

Reply:
I just had a chat with the Cadwyn Author  ( @zmievsa ) who recommends to only add it for breaking changes.

> Depending on your needs. My general recommendation is to only add migrations for breaking changes 
https://docs.cadwyn.dev/how_to/change_endpoints/#add-a-new-endpoint

Reply:
I'll make Cadwyn's docs more verbose on when it makes the most sense to add a migration. Concepts section mostly focuses on what's possible with Cadwyn while "how to" focuses on what you should actually do. 

Either way 99% of the time it makes sense to add an endpoint to all versions since it's not a breaking change :)